wordpress-mobile/WordPress-iOS
# My Site Dashboard: Custom feature announcement

Closes #18209

## Description
- Refactored `WhatIsNewView` to depend on an appearance object that gets injected when creating the view.
- Created a new announcement cell to match the new design
- Used `appVersionName` to determine if the announcement is for the dashboard
- Extracted the view's title from the returned features. 
- Injected custom appearance and new custom cell when showing announcement for dashboard  

/ | Light | Dark
-- | -- | --
Dashboard | <img src="https://user-images.githubusercontent.com/25306722/160506159-924b1970-052b-4eed-806f-e41c7158d926.png"> | <img src="https://user-images.githubusercontent.com/25306722/160506175-e651abcd-e88e-41e2-8b64-85c3ad0dd567.png">
Default | <img src="https://user-images.githubusercontent.com/25306722/160507808-a950b9ee-6d26-41c5-beb2-b0d39b40d38c.png"> | <img src="https://user-images.githubusercontent.com/25306722/160507787-5e3e5900-69e4-4d1c-84e3-0df534578168.png">

## Testing Instructions
First you need to modify the response of the feature announcement API to return custom response for testing. This can be achieved in multiple ways. Here are two options:

<details>
  <summary><strong>Hardcoding response in code (Easier)</strong></summary>
  <br>

  - Replace `AnnouncementServiceRemote.swift` in WordPressKit with this [version](https://drive.google.com/file/d/1HUtBzTB7vdDjNhwXaAMPZERBD9JX0rZ9/view?usp=sharing) (This file is in an external pod. Make sure to revert the changes after you're done by removing the pod from Pods folder and running `bundle exec pod install`)
  - Replace `AnnouncementsStore.swift` by this [version](https://drive.google.com/file/d/1RZ4xae57r3XFnHo-kd5ciaX9PO_l1ehj/view?usp=sharing)
  ---
</details>
<details>
  <summary><strong>Using Proxy Apps</strong></summary>
  <br>

  - Install [ProxyMan](https://proxyman.io)
  - Configure ProxyMan to [capture simulator traffic](https://docs.proxyman.io/debug-devices/ios-simulator)   
  - Make sure the app's network traffic shows up in ProxyMan
  - Add a local map for the targetted API
      - Go to Tools -> Map Local...
      - Add new mapping
      - Set URL to `https://public-api.wordpress.com/wpcom/v2/mobile/feature-announcements`
      - Make sure to check "Include all subpaths of this URL"
      - Under "Map to" choose Local Directory and use this [file](https://drive.google.com/file/d/1McI4p5Gl-Ew59Ql7mngpwSPoLVp9PRyh/view?usp=sharing)
  ---
</details>

_PS: Once I have access to the Mobile Announce mc tool, I will create an announcement targeting this PR, which should remove the need for the above requirement._

1. Make sure to follow one of the above options
2. To ease the testing modify `WhatIsNewScenePresenter.shouldPresentWhatIsNew` to always return true
3. Run the app
4. The announcement should be displayed and should match the design
5. Change the `appVersionName` in the response to anything that's not "19.6"
6. Rerun the app
7. The announcement should be displayed and should match the original design
8. Go to My Site
9. Tap the FAB then tap Story Post
10. Make sure Stories intro is displayed correctly
11. Please test light and dark mode

## Regression Notes
1. Potential unintended areas of impact
Normal feature announcement view and stories intro

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually the showing of normal announcements

12. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
